lib/debug/breakpoint.rb

```diff
@@ -343,7 +346,10 @@ def watch_eval
         begin
           @prev = @current
           @current = result
-          suspend
+
+          if @cond.nil? || @object.instance_eval(@cond)
```

Comment:
IMO, the if condition of WatchBreakpoint should also be evaluated on the target object.